pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -613,6 +609,51 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 	}
 }
 
+// update the node associated with the connection, after receiving a a packet from envoy.
+func (s *DiscoveryServer) initConnectionNode(discReq *xdsapi.DiscoveryRequest, con *XdsConnection) error {
+	con.mu.RLock()
+	if con.modelNode != nil {
+		con.mu.RUnlock()
+		return nil // only need to init the node on first request in the stream
+	}
+	con.mu.RUnlock()
```

Comment:
AFAIK access to a pointer is not atomic - it needs to be synced if another goroutine may modify the pointer while it is read.  In this case it may not be needed ( since con.modelNode is set at connect ), but I would rather have it.